src/core/router/info.cc

```diff
@@ -139,99 +59,96 @@ RouterInfo::~RouterInfo()
 }
 
 RouterInfo::RouterInfo(const std::string& path)
-    : m_Path(path), m_Buffer(std::make_unique<std::uint8_t[]>(Size::MaxBuffer))  // TODO(anonimal): buffer refactor
+    : m_Path(path),
+      m_Buffer(std::make_unique<std::uint8_t[]>(Size::MaxBuffer)),  // TODO(anonimal): buffer refactor
+      m_Exception(__func__)
 {
-  if (!ReadFromFile())
-    throw std::runtime_error("RouterInfo: invalid file");
+  ReadFromFile();
   ReadFromBuffer(false);
 }
 
 RouterInfo::RouterInfo(const std::uint8_t* buf, std::uint16_t len)
     : m_Buffer(std::make_unique<std::uint8_t[]>(Size::MaxBuffer)),  // TODO(anonimal): buffer refactor
-      m_BufferLen(len)
+      m_BufferLen(len),
+      m_Exception(__func__)
 {
   if (!buf)
     throw std::invalid_argument("RouterInfo: null buffer");
   if (len < Size::MinBuffer || len > Size::MaxBuffer)
     throw std::length_error("RouterInfo: invalid buffer length");
-  memcpy(m_Buffer.get(), buf, Size::MaxBuffer);
+  std::memcpy(m_Buffer.get(), buf, Size::MaxBuffer);
```

Comment:
Yep, thanks. I kept for continuity since this would be fixed by the buffer refactor (noted in TODO). Since you're working with #635, I'll make this change now.